src/type_conversions/rpc/trace.rs

```diff
@@ -225,64 +234,121 @@ impl ToEthers<EthersTrace> for LocalizedTransactionTrace {
                 EthersAction::Suicide(_) => EthersActionType::Suicide,
                 EthersAction::Reward(_) => EthersActionType::Reward,
             },
-            error: match self.trace.result {
-                Some(TraceResult::Error { error }) => Some(error),
-                _ => None,
-            },
+            error: self.trace.error,
+        }
+    }
+}
+
+// -----------------------------------------------
+
+impl ToReth<CallOutput> for EthersCallResult {
+    fn into_reth(self) -> CallOutput {
+        CallOutput { gas_used: self.gas_used.into_reth(), output: self.output.into_reth() }
+    }
+}
+
+impl ToEthers<EthersCallResult> for CallOutput {
+    fn into_ethers(self) -> EthersCallResult {
+        EthersCallResult {
+            gas_used: self.gas_used.into_ethers(),
+            output: self.output.into_ethers(),
         }
     }
 }
 
 // -----------------------------------------------
 
-/// EthersTrace (ethers) -> LocalizedTransactionTrace (reth)
-impl ToReth<TraceResult> for EthersRes {
-    fn into_reth(self) -> TraceResult {
+impl ToReth<CreateOutput> for EthersCreateResult {
+    fn into_reth(self) -> CreateOutput {
+        CreateOutput {
+            gas_used: self.gas_used.into_reth(),
+            code: self.code.into_reth(),
+            address: self.address.into_reth(),
+        }
+    }
+}
+
+impl ToEthers<EthersCreateResult> for CreateOutput {
+    fn into_ethers(self) -> EthersCreateResult {
+        EthersCreateResult {
+            gas_used: self.gas_used.into_ethers(),
+            code: self.code.into_ethers(),
+            address: self.address.into_ethers(),
+        }
+    }
+}
+
+// -----------------------------------------------
+
+impl ToReth<Option<TraceOutput>> for EthersRes {
+    fn into_reth(self) -> Option<TraceOutput> {
         match self {
-            EthersRes::Call(EthersCallResult { gas_used, output }) => {
-                TraceResult::parity_success(TraceOutput::Call(CallOutput {
-                    gas_used: gas_used.into_reth(),
-                    output: output.into_reth(),
-                }))
-            }
-            EthersRes::Create(EthersCreateResult { gas_used, code, address }) => {
-                TraceResult::parity_success(TraceOutput::Create(CreateOutput {
-                    gas_used: gas_used.into_reth(),
-                    code: code.into_reth(),
-                    address: address.into_reth(),
-                }))
-            }
-            EthersRes::None => TraceResult::Error { error: "Error".to_string() },
+            EthersRes::Call(result) => Some(TraceOutput::Call(result.into_reth())),
+            EthersRes::Create(result) => Some(TraceOutput::Create(result.into_reth())),
+            EthersRes::None => None,
```

Comment:
Implementing this as `ToReth<Option<TraceOutput>>` instead of `ToReth<TraceOutput>` is a bit of a hack, because `TraceOutput ` does not have an equivalent of `EthersRes::None`.